code/procs/gamehelpers.dm

```diff
@@ -350,21 +350,23 @@ var/obj/item/dummy/click_dummy = new
 	return message
 
 
-/proc/can_see(var/atom/source, var/atom/target, var/length=5) // I couldnt be arsed to do actual raycasting :I This is horribly inaccurate.
+/proc/can_see(atom/source, atom/target, length=5) // I couldnt be arsed to do actual raycasting :I This is horribly inaccurate.
 	var/turf/current = get_turf(source)
 	var/turf/target_turf = get_turf(target)
-	var/steps = 0
-
-	while(current != target_turf)
-		if(steps > length) return 0
-		if(!current) return 0
-		if(current.opacity) return 0
+	if(current == target_turf)
+		return TRUE
+	if(get_dist(current, target_turf) > length)
+		return FALSE
+	current = get_step_towards(source, target_turf)
+	while((current != target_turf))
+		if(current.opacity)
+			return FALSE
 		for(var/atom/A in current)
-			if(A.opacity) return 0
+			if(A.opacity) 
+				return FALSE
 		current = get_step_towards(current, target_turf)
```

Comment:
Making sure - is `current = get_step_towards(current, target_turf)` supposed to be here twice?

Reply:
That one yes, the other one may be able to be cut but I don't know if it'll break the loop so uhh I'll do some tests